botocore/response.py

```diff
@@ -80,6 +80,21 @@ def read(self, amt=None):
             self._verify_content_length()
         return chunk
 
+    def readlines(self, delimiter='\n'):
+        """Generator that streams lines. The default delimiter
+        is the newline character.
+
+        If the delimiter argument is specified, it will
+        override the default.
+        """
+        line_buffer = ''
+        for byte in self.read():
+            byte = line_buffer + str(byte)
```

Comment:
This is creating a new copy of a str each time and creates O(n^2) behavior).  What about reading contents in a fixed chunk size (say 8/16k) and then splitting the lines for each buffer?